app/src/main/java/org/wikipedia/notifications/PollNotificationService.kt

```diff
@@ -0,0 +1,106 @@
+package org.wikipedia.notifications
+
+import android.app.job.JobInfo
+import android.app.job.JobParameters
+import android.app.job.JobScheduler
+import android.app.job.JobService
+import android.content.ComponentName
+import android.content.Context
+import io.reactivex.rxjava3.schedulers.Schedulers
+import kotlinx.coroutines.CoroutineExceptionHandler
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.launch
+import org.wikipedia.WikipediaApp
+import org.wikipedia.csrf.CsrfTokenClient
+import org.wikipedia.database.AppDatabase
+import org.wikipedia.dataclient.ServiceFactory
+import org.wikipedia.dataclient.WikiSite
+import org.wikipedia.dataclient.mwapi.MwException
+import org.wikipedia.settings.Prefs
+import org.wikipedia.util.log.L
+
+class PollNotificationService : JobService() {
+
+    private var jobParameters: JobParameters? = null
+
+    override fun onStartJob(jobParameters: JobParameters?): Boolean {
+        pollNotifications(this)
+        return true
+    }
+
+    override fun onStopJob(p0: JobParameters?): Boolean {
+        jobParameters = p0
```

Comment:
Shouldn't we assign `jobParameters` when the job is started, instead of stopped?